linux/function_models/libz_wrapper.c

```diff
@@ -31,6 +31,10 @@
 
 uint32_t crc32(uint32_t crc, const uint8_t *buf, unsigned len) {
     if (!g_enable_function_models) {
+        if (!orig_crc32) {
```

Comment:
Would it be possible to turn these functions into a template? It would reduce the amount of code duplication.
We'd have to switch this file to c++, I hope that would work.

Reply:
Sure, I'll look into it.